### DISCUSSION TITLE
Problem connecting to mongoDB

### DISCUSSION THREAD

**Discussion by yarin-bcure** [getredash/redash#6629] [Q&A]
Hi all.
I installed Redash as a local application on my Ubuntu machine (I entered Redash with the http://localhost:5001 URL). The application is working fine, and I even managed to connect a MySQL data source.

Now I am trying to connect an AWS documentDB instance (located under a cluster) using the MongoDB data source and having trouble doing it (keep getting timeout). The connection string I am using is corresponds to the template: `mongodb://user:pass@host:port/db_name?options`. I also write the DB name in the `Database name` field.

- I can connect to the DB using nosqlbooster with this corresponding template: `mongodb://user:pass@host:port/?options`.

Can someone help me with the configuration or at least give me an example of the complete configuration of the MongoDB?

**Comment by masayuki038** [comment]
Hi @yarin-bcure, I haven't connected to AWS DocumentDB...
According to [this page](https://docs.aws.amazon.com/documentdb/latest/developerguide/connect_programmatically.html), it seems the settings for connecting to it depend on enabling TLS or not. There are some instructions and sample programs in it. Could you check it?

Redash uses `pymongo` as well. If you can connect using the sample program introduced on the page, I think you can also connect from Redash. Please try.
